### PR TITLE
Remove TheJobTriggerType and improve the internal job logic

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
What has changed:
* The enum `TheJobTriggerType` is no there anymore. (Fixes: #880 )
* The handlers are event independent. We don't have build handlers for each trigger.
* There are no handlers for comments specifically.
* Each handler explicitly states:
    * What event it can react on.
    * What `job_type` it uses in the config.
    * What command you can use in a comment to trigger it.

The workflow for handling events has been simplified:
* Each event has a database trigger (Like PRModel, BranchModel,...) attached. (Created or existing one.)
* For database trigger, we know matching `JobConfigTrigger` (the enum user can specify).
* For each handler, we know, which `JobType` it matches (the enum user can specify). e.g. `CoprBuildHandler`, `CoprBuildEnd`, `CoprBuildStart` matches both `build` and `copr-build`. And wise-versa, for each `JobType`, we know all the handlers.
* The process works in the following way: **We search through the configured jobs, filter those with `JobConfigTrigger` matching event and pick all the handlers that matches `JobType` and can react to the event.**

TODO:
- [x] Fix the tests. (Clear the existing ones, create new ones.)
- [x] (Re)add the pre-checks from removed handlers.
- [x] (Re)implement require jobs.
- [x] (Re)implement issue handling. (Propose-downstream only for now.)

Follow-ups (to not make this too big):
- Merge `get_handlers_for_event` and `get_config_for_handler_kls` to return pair of handler, job directly.
- Support for arguments in comment commands.
- Reacting to issues -- hardcoded to propose-update for now. (But will be much easier to do now.)
- Various refactorings, renames,... -- It would make the PR much bigger..;)

### CODE REVIEW

**Comment by jpopelka** [comment]
I'll give you thumbs up once it's not WIP, but for now, it's too-huge-to-review :-)

**Comment by lachmanfrantisek** [comment]
Rebased and reworded commit messages.

**Comment by lachmanfrantisek** [comment]
The only (probably unrelated) failure:

```

=================================== FAILURES ===================================
_____________________________ test_get_api_client ______________________________

    def test_get_api_client():
        """ let's make sure we can get k8s API client """
>       assert sandcastle.Sandcastle.get_api_client()

tests_requre/openshift_integration/test_sandcastle.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    @staticmethod
    def get_api_client() -> CoreV1Api:
        """
        Obtain API client for kubenernetes; if running in a pod,
        load service account identity, otherwise load kubeconfig
        """
        logger.debug("Initialize kubernetes client")
        configuration = Configuration()
        if "KUBERNETES_SERVICE_HOST" in os.environ:
            logger.info("loading incluster config")
>           load_incluster_config(client_configuration=configuration)
E           TypeError: load_incluster_config() got an unexpected keyword argument 'client_configuration'
```

File: packit_service/worker/handlers/abstract.py [hunk 1]
```diff
@@ -219,29 +229,18 @@ def __init__(
...
-            ):
-                self._db_trigger = PullRequestModel.get_by_id(self.data.trigger_id)
-            elif self.data.trigger == TheJobTriggerType.issue_comment:
-                self._db_trigger = IssueModel.get_by_id(self.data.trigger_id)
-            elif self.data.trigger == TheJobTriggerType.release:
-                self._db_trigger = ProjectReleaseModel.get_by_id(self.data.trigger_id)
-            elif self.data.trigger == TheJobTriggerType.push:
-                self._db_trigger = GitBranchModel.get_by_id(self.data.trigger_id)
-        return self._db_trigger
-
```
Comment (TomasTomecek) [inline packit_service/worker/handlers/abstract.py:238]: nice!

File: packit_service/service/events.py [hunk 2]
```diff
@@ -100,21 +97,6 @@ class WhitelistStatus(enum.Enum):
     approved_manually = WHITELIST_CONSTANTS["approved_manually"]
 
 
-class TheJobTriggerType(str, enum.Enum):
```
Comment (TomasTomecek) [inline packit_service/service/events.py:103]: you did it :)

**Review by TomasTomecek — APPROVED** [review]
code LGTM, I'd like to see more docs for the dependency-related decorators since it's not super-clear to me how would I use those - feel free to do that in a new PR so you don't need to rebase this, hence the approve

**Comment by lachmanfrantisek** [comment]
>  I'd like to see more docs for the dependency-related decorators since it's not super-clear to me how would I use those - feel free to do that in a new PR so you don't need to rebase this, hence the approve

I'll send a few follow-ups.  (And will include this there.)

File: packit_service/worker/handlers/abstract.py [hunk 3]
```diff
@@ -40,56 +40,76 @@
...
+        MAP_REQUIRED_JOB_TYPE_TO_HANDLER[job_type].add(kls)
+        return kls
+
+    return _add_to_mapping
+
+
+def reacts_to(event: Type["Event"]):
+    """
+    [class decorator]
+    Specify an eventr for which we want to use this handler.
```
[the remark this answers is not in the document]
Comment (lachmanfrantisek) [reply packit_service/worker/handlers/abstract.py:93]: Resolved in #915

**Pull request merged**